### PR TITLE
Adding NVIDIA GPUNet

### PR DESCRIPTION
**Pull request by nv-kkudrynski**
Hi. Adding another NVIDIA model  - GPUNet. Notebook tested on our side.

### CODE REVIEW

File: nvidia_deeplearningexamples_gpunet.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+if torch.cuda.is_available():
+    !nvidia-smi
+else:
+    torch.device("cpu")
+```
+
+### Load Pretrained model
+Loads NVIDIA GPUNet-0 model by default pre-trained on IMAGENET dataset. You can switch the default pre-trained model loading from GPUNet-0 to one of the following models listed below. 
+
+The model architecture is visible as output of the loaded model. For details architecture and latency info please refer to Figure#[6](https://drive.google.com/file/d/1pLA0OopYVRpYreIhQQKPdNauaaBEGITe/view) and Table#[3](https://drive.google.com/file/d/1pLA0OopYVRpYreIhQQKPdNauaaBEGITe/view) in the CVPR-2022 paper.
```
Comment (vmoens) [inline nvidia_deeplearningexamples_gpunet.md:58]: I personally don't have access to these figures

File: nvidia_deeplearningexamples_gpunet.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+| :----- | :----- |
+| `GPUNet-0` | GPUNet-0 has the fastest measured latency on GV100 |
+| `GPUNet-1` | GPUNet-1 has improved accuracy with one additional layer on GPUNet-0|
+| `GPUNet-2` | GPUNet-2 has higher accuracy with two additional layers on GPUNet-0 |
+| `GPUNet-P0` | GPUNet-P0 is the distilled model with higher accuracy than GPUNet-0 but similar latency|
+| `GPUNet-P1` | GPUNet-P1 is distilled model with even higher accuracy than GPUNet-1 but similar latency |
+| `GPUNet-D1` | GPUNet-D1 has the second highest accuracy amongst all GPUNets|
+| `GPUNet-D2` | GPUNet-D2 has the highest accuracy amongst all GPUNets |
+
+***Note***: Loaded model defaults to fp32 weights. To switch to fp16 gpunet model weights, set parameter ***model_math='fp16'***.
```
Comment (vmoens) [inline nvidia_deeplearningexamples_gpunet.md:72]: not sure if that breaks the CI but `model_math='fp16'` is perhaps a bit neater?

File: nvidia_deeplearningexamples_gpunet.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    [utils.prepare_input_from_uri(uri) for uri in uris]
+).to(device)
+
+print("Ready to run inference...")
+```
+
+### Run inference
+Use `pick_n_best(predictions=output, n=topN)` helper function to pick N most probable hypotheses according to the model.
+
+***Note***: If using fp16 gpunet model weights replace gpunet(batch) with ***gpunet(batch.half())***.
```
Comment (vmoens) [inline nvidia_deeplearningexamples_gpunet.md:101]: same let's use ` decorators for this unless there's a reason not to

File: nvidia_deeplearningexamples_gpunet.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    img = Image.open(requests.get(uri, stream=True).raw)
+    img.thumbnail((256,256), Image.ANTIALIAS)
+    plt.imshow(img)
+    plt.show()
+    print(result)
+```
+
+### Details
+For detailed information on model input and output, training recipies, inference and performance visit:
+[github](https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Classification/GPUNets)
```
Comment (vmoens) [inline nvidia_deeplearningexamples_gpunet.md:121]: the link should be https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Classification/GPUNet I think

**Review by vmoens — COMMENTED** [review]
Brilliant, as always!
See my comments then we're good to go!

File: nvidia_deeplearningexamples_gpunet.md (hunk 2 above)
Comment (nv-kkudrynski) [reply nvidia_deeplearningexamples_gpunet.md:72]: im not sure if i understand. you mean make fp16 the default?

Comment (vmoens) [reply nvidia_deeplearningexamples_gpunet.md:72]: I mean using ` decorators instead of *

Comment (nv-kkudrynski) [reply nvidia_deeplearningexamples_gpunet.md:72]: Ha. Communication! :) For the last hour we were trying to wrap model object with fp16 halving **decorator** :P That's not the best idea btw, but anyway next commit will make the code much more clear :)

Comment (vmoens) [reply nvidia_deeplearningexamples_gpunet.md:72]: My bad should have made my thinking clearer, i tend to assume everyone has an open access to my thoughts :)

**Comment by nv-kkudrynski** [comment]
@vmoens I will have a new commit soon to address all your comments. In the meanwhile can you please make the CLA Check pass for me? This should be fixed long time ago. I emailed cla@fb.com with all the agreements.

File: nvidia_deeplearningexamples_gpunet.md (hunk 1 above)
Comment (nv-kkudrynski) [reply nvidia_deeplearningexamples_gpunet.md]: Fixed

File: nvidia_deeplearningexamples_gpunet.md (hunk 2 above)
Comment (nv-kkudrynski) [reply nvidia_deeplearningexamples_gpunet.md]: Fixed

File: nvidia_deeplearningexamples_gpunet.md (hunk 4 above)
Comment (nv-kkudrynski) [reply nvidia_deeplearningexamples_gpunet.md]: Fixed

**Review by vmoens — APPROVED** [review]

**Comment by nv-kkudrynski** [comment]
Yes. Seems fixed now. Thanks for your support @vmoens
